src/Counter.elm

```diff
@@ -0,0 +1,97 @@
+module Counter exposing (main)
+
+import Browser
+import Html exposing (button, div, h2, input, strong, text)
+import Html.Attributes as Attributes
+import Html.Events exposing (onClick, onInput)
+
+
+main =
+    Browser.sandbox
+        { init = init
+        , update = update
+        , view = view
+        }
+
+
+type alias Model =
+    { counter : Float
+    , content : String
+    , lastCounter : Float
+    }
+
+
+init : Model
+init =
+    { counter = 0, content = "0", lastCounter = 0 }
+
+
+type Msg
+    = Plus
+    | Minus
+    | Multiply
+    | Div
+    | ChangeNumber String
+    | ResetCounter
+    | Undo
+
+
+update : Msg -> Model -> Model
+update msg model =
+    let
+        userInputNum =
+            Maybe.withDefault 0 (String.toFloat model.content)
```

Comment:
Prefer pipes (`<|`) over trailing parentheses.

Reply:
For this specific use case, you should use [`Maybe.Extra.unwrap`](https://package.elm-lang.org/packages/elm-community/maybe-extra/latest/Maybe-Extra#unwrap):
```elm
Maybe.Extra.unwrap 0 String.toFloat model.content
```